src/parser/csql_grammar.y

```diff
@@ -5185,24 +5192,74 @@ only_all_class_spec
 		DBG_PRINT}}
 	;
 
-class_name
-	: identifier DOT identifier
+user_specified_name_without_dot
+	: identifier_without_dot DOT identifier_without_dot
 		{{
 
-			PT_NODE *user_node = $1;
-			PT_NODE *name_node = $3;
+			PT_NODE *user = $1;
+			PT_NODE *name = $3;
 
-			if (name_node != NULL && user_node != NULL)
+			if (user && name)
 			  {
-			    name_node->info.name.resolved = pt_append_string (this_parser, NULL,
-			                                                      user_node->info.name.original);
+			    name->info.name.resolved = pt_append_string (this_parser, NULL, user->info.name.original);
+
+			    parser_free_tree (this_parser, user);
+
+				// PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
 			  }
-			if (user_node != NULL)
+
+			$$ = name;
+			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
+
+		DBG_PRINT}}
+	| identifier_without_dot
+		{{
+
+			PT_NODE *name = $1;
+
+			if (name)
```

Comment:
This if block should be removed.

Reply:
The code for the PT_NAME_INFO_USER_SPECIFIED flag didn't come with it, so the code below is commented out.
```
PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
```
Since it is a feature branch, the code for the flag will be included through another PR. So keep the block.